parallel/slice.go

```diff
@@ -1,98 +1,56 @@
 package parallel
 
-import "sync"
+import (
+	"runtime"
+	"sync"
+)
 
 // Map manipulates a slice and transforms it to a slice of another type.
-// `iteratee` is call in parallel. Result keep the same order.
+// `iteratee` is called in parallel. Result keep the same order.
 func Map[T any, R any](collection []T, iteratee func(T, int) R) []R {
 	result := make([]R, len(collection))
-
-	var wg sync.WaitGroup
-	wg.Add(len(collection))
-
-	for i, item := range collection {
-		go func(_item T, _i int) {
-			res := iteratee(_item, _i)
-
-			result[_i] = res
-
-			wg.Done()
-		}(item, i)
-	}
-
-	wg.Wait()
-
+	NewTaskPool(collection, runtime.NumCPU()/2, func(v T, i int) {
```

Comment:
Why did you hard code the pool size instead of using a parameter to the function? 🤔

A default value could be useful as well:

```go
var DefaultPoolSize = runtime.NumCPU()/2
```

I'm also thinking about the case of i/o bound jobs. The number of parallel jobs won't depend on number of CPU ;)

Reply:
Done